### PR TITLE
Create 1.7.2 patch release

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 
Create 1.7.2 patch release

Please click each commit id to see more information. They are cherry-picked from the master branch. 

It is just some cherry-picks to make the code still compliable. 
**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Review by smk2007 — APPROVED** [review]

**Comment by snnn** [comment]
At least 2 approving reviews are required for this PR.

**Comment by snnn** [comment]
Will need to cherry-pick #7140 in.

File: onnxruntime/core/providers/cuda/tensor/resize_impl.cu [hunk 1]
```diff
@@ -10,29 +10,29 @@ using onnxruntime::UpsampleMode;
...
   }
 }
 
 __device__ int NearestPixel_ROUND_PREFER_FLOOR(float x_original, bool) {
   if (x_original == static_cast<int>(x_original) + 0.5f) {
-    return static_cast<int>(floor(x_original));
+    return static_cast<int>(_Floor(x_original));
   }
-  return static_cast<int>(round(x_original));
+  return static_cast<int>(roundf(x_original));
```
Comment (snnn) [inline onnxruntime/core/providers/cuda/tensor/resize_impl.cu:23]: I changed std::round to roundf. The behavior should be the same.

**Review by askhade — APPROVED** [review]

**Pull request merged**